koku/masu/processor/_tasks/data_validation.py

```diff
@@ -0,0 +1,222 @@
+                        AND lpad(month, 2, '0') = '{month}'
+                        GROUP BY {query_filters.get("date")}
+                        ORDER BY {query_filters.get("date")}"""
+                result = report_db_accessor._execute_trino_raw_sql_query(sql, log_ref="data validation query")
+            else:
+                source = "source_uuid" if not self.ocp_on_cloud_type else "cluster_id"
+                sql = f"""
+                    SELECT sum({query_filters.get("metric")}) as metric, {query_filters.get("date")} as date
+                    FROM {self.schema}.{table}_{year}_{month}
+                        WHERE {source} = '{provider_filter}'
+                        AND {query_filters.get("date")} >= '{start}'
+                        AND {query_filters.get("date")} <= '{end}'
+                        GROUP BY {query_filters.get("date")}
+                        ORDER BY {query_filters.get("date")}"""
+                result = report_db_accessor._prepare_and_execute_raw_sql_query(
+                    table, sql, operation="VALIDATION_QUERY"
+                )
+            if result != []:
+                for day in result:
+                    key = day[1].date() if isinstance(day[1], datetime) else day[1]
+                    daily_result[key] = float(day[0])
+        return daily_result
+
+    def check_data_integrity(self):
+        """Helper to call the query and validation methods for validating data"""
+        valid_cost = False
+        pg_data = None
+        trino_data = None
+        cluster_id = None
+        daily_difference = {}
+        LOG.info(log_json(msg="validation started for provider", context=self.context))
+        provider = Provider.objects.filter(uuid=self.provider_uuid).first()
+        provider_type = provider.type.strip("-local")
+        if self.ocp_on_cloud_type:
+            provider_type = self.ocp_on_cloud_type.strip("-local")
+            cluster_id = provider.authentication.credentials.get("cluster_id")
+        # Postgres query to get daily values
+        try:
+            pg_data = self.execute_relevant_query(provider_type, cluster_id)
+        except Exception as e:
+            LOG.warning(log_json(msg=f"data validation postgres query failed: {e}", context=self.context))
+            return
+        # Trino query to get daily values
+        try:
+            trino_data = self.execute_relevant_query(provider_type, cluster_id, True)
+        except Exception as e:
+            LOG.warning(log_json(msg=f"data validation trino query failed: {e}", context=self.context))
+            return
+        # Compare results
+        LOG.debug(f"PG: {pg_data} Trino data: {trino_data}")
+        daily_difference, valid_cost = self.compare_data(pg_data, trino_data)
+        if valid_cost:
+            LOG.info(log_json(msg=f"all data complete for provider: {self.provider_uuid}", context=self.context))
+            provider.data_valid = True
+        else:
+            LOG.warning(
```

Comment:
We could raise here or log an error instead if we want.

Reply:
IMO we should raise an error if the data is not valid, it will bring more awareness to times when these don't line up and it should be an actionable event for someone to determine why the costs aren't the same.